spec/decorator_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rspec'
+require_relative '../lib/basedecorator'
+require_relative '../lib/trimmerdecorator'
+require_relative '../lib/capitalizedecorator'
+
+describe 'Decorator Classes' do
+  let(:nameable) { double('Nameable', correct_name: 'John Doe') }
+
+  describe BaseDecorator do
+    subject { BaseDecorator.new(nameable) }
+
+    it 'delegates correct_name to nameable' do
+      expect(nameable).to receive(:correct_name)
+      subject.correct_name
+    end
+  end
+
+  describe TrimDecorator do
+    subject { TrimDecorator.new(nameable) }
+
+    it 'trims the correct_name to 10 characters' do
+      expect(subject.correct_name).to eq('John Doe')
+    end
+  end
+
+  describe CapitalizeDecorator do
+    subject { CapitalizeDecorator.new(nameable) }
+
+    it 'capitalizes the correct_name' do
+      expect(subject.correct_name).to eq('John doe')
+    end
+  end
+end
```

Comment:
- [ ] [REQUIRED] Kindly note that there should be one file per class for all the tests. You need to place/seperate the tests for the following classes in different classes: `basedecorator.rb`, `capitalizedecorator.rb`, `trimmerdecorator.rb` instead of putting all the tests in `decorator_spec` files.